e2e/kind.sh

```diff
@@ -189,6 +192,19 @@ test_reject_peer_empty_public_key() {
 	assert_fail "create_peer e2e 10.5.0.1/32 0 ''" "should not be able to create Peer with empty public key"
 }
 
+test_fullmesh_allowed_location_ips() {
+	docker exec kind-cluster-kilo-control-plane ip address add 10.6.0.1/32 dev eth0
+	$KUBECTL_BINARY annotate node kind-cluster-kilo-control-plane kilo.squat.ai/allowed-location-ips=10.6.0.1/32
+	docker exec kind-cluster-kilo-worker ip address add 10.6.0.2/32 dev eth0
+	$KUBECTL_BINARY annotate node kind-cluster-kilo-worker kilo.squat.ai/allowed-location-ips=10.6.0.2/32
+	assert_equals Unauthorized "$(retry 10 5 'IP is not yet routable' curl_pod -m 1 -s -k https://10.6.0.1:10250/healthz)" "should be able to make HTTP request to allowed location IP"
+	assert_equals Unauthorized "$(retry 10 5 'IP is not yet routable' curl_pod -m 1 -s -k https://10.6.0.2:10250/healthz)" "should be able to make HTTP request to allowed location IP"
+	$KUBECTL_BINARY annotate node kind-cluster-kilo-control-plane kilo.squat.ai/allowed-location-ips-
```

Comment:
if we take off the annotations, we might as well test if pinging the allowed IPs stops working. It should also not work in the case where the curl pod runs on the same node, right?

Reply:
yes, that's a very nice idea!

Reply:
grr my testing shows that the 10.6.0.X IPs are still accessible when the curl helper pod is on the same node as the IP. This makes sense to me since we are not blocking accessing IPs Pods even if we haven't explicitly been made aware of that IP; for example, a Pod can access `google.com` even though we haven't added that IP explicitly to a list of allowed IPs for that node.

In other words, this will always work implicitly when the Pod is located on the same node as the allowed IP; the goal of the Kilo feature was to make this explicit and allow nodes in other locations to access this too.

One way around this would be to always ensure that the curl helper Pod is on a node without access to the extra IP. To do this, we could add a scheduling rule to ensure the Pod is always on a control-plane node for example.